clients/js/genjs.go

```diff
@@ -213,17 +226,27 @@ var methodTmplStr = `
           if (retries < backoffs.length && retryPolicy.retry(requestOptions, err, response, body)) {
             const backoff = backoffs[retries];
             retries += 1;
-            return setTimeout(requestOnce, backoff);
+            setTimeout(requestOnce, backoff);
+            return;
           }
           if (err) {
-            return rejecter(err);
+            rejecter(err);
+            return;
           }
-          if (response.statusCode >= 400) {
-            return rejecter(new Error(body));
+          switch (response.statusCode) {
+            {{ range $response := .Responses }}case {{ $response.StatusCode }}:{{if $response.IsError }}
+              rejecter(new Errors.{{ $response.Name }}(body || {}));
+            {{else}}{{if $response.IsNoData}}
+              resolver();
+            {{else}}
+              resolver(body);
+            {{end}}{{end}}  break;
+            {{end}}default:
+              rejecter(new Error("Recieved unexpected statusCode " + response.statusCode));
```

Comment:
in go we turn this into an internal error `new InternalError(...`, not sure if we want to do that in JS too